src/main/java/org/folio/bulkops/processor/HoldingsDataProcessor.java

```diff
@@ -188,9 +202,17 @@ public Class<ExtendedHoldingsRecord> getProcessedType() {
     return ExtendedHoldingsRecord.class;
   }
 
-  private boolean ruleTenantsAreNotValid(BulkOperationRule rule, Action action, ExtendedHoldingsRecord extendedHolding) {
+  private boolean ruleTenantsAreNotValid(BulkOperationRule rule, Action action, UpdateOptionType option, ExtendedHoldingsRecord extendedHolding) {
     var ruleTenants = rule.getRuleDetails().getTenants();
     var actionTenants = action.getTenants();
+    if (nonNull(ruleTenants) && !ruleTenants.isEmpty() && nonNull(actionTenants) && !actionTenants.isEmpty()) {
```

Comment:
Hi @obozhko-folio , it is better to have here separate method to understand why such checking is needed.

Reply:
Fixed.